server/src/option.rs

```diff
@@ -104,6 +104,37 @@ impl Config {
         validate_path_is_writeable(staging_path)
     }
 
+    pub async fn validate_storage(&self) -> Result<(), ObjectStorageError> {
+        let obj_store = self.storage.get_object_store();
+        let rel_path = relative_path::RelativePathBuf::from(".parseable.json");
+
+        let has_parseable_json = obj_store.get_object(&rel_path).await.is_ok();
+
+        let has_dirs = match obj_store.list_dirs_in_storage().await {
+            Ok(dirs) => dirs.is_empty(),
+            Err(_) => false,
+        };
+
+        let has_streams = obj_store.list_streams().await.is_ok();
+
+        if has_dirs || has_parseable_json && has_streams {
+            Ok(())
+        } else if has_parseable_json && !has_streams {
+            Err(ObjectStorageError::Custom(
+                "Parseable config found, but the Storage contains some stale data.".to_owned(),
+            ))
+        } else if !has_parseable_json && !has_streams && !has_dirs {
+            Err(ObjectStorageError::Custom(
+                "Storage contains some stale data, Please provide an Empty Storage".to_owned(),
+            ))
+        } else {
+            Err(ObjectStorageError::Custom(
+                "Parseable config is missing, but streams are present in Storage.\nJoin us on Parseable Slack"
```

Comment:
Please provide the scenario when can this happen

Reply:
This is again a highly unlikely scenario to happen. 
This error will be triggered if `.parseable.json` was deleted from the storage manually.